packages/lodestar/src/network/network.ts

```diff
@@ -99,6 +100,39 @@ export class Libp2pNetwork extends (EventEmitter as { new(): NetworkEventEmitter
     await this.libp2p.hangUp(peerInfo);
   }
 
+  // peerIds: peers already have this subnet
+  public async connectToNewPeersBySubnet(subnet: number, inPeerIds?: string[]): Promise<number> {
+    const peerIds = inPeerIds || [];
+    // @ts-ignore
+    const discovery: Discv5Discovery = this.libp2p._discovery.get("discv5");
+    const discv5: Discv5 = discovery.discv5;
+    const peerInfosForSubnet: PeerInfo[] = await Promise.all(
+      discv5.kadValues()
+        .filter((enr: ENR) => enr.get("attnets"))
+        .filter((enr: ENR) => this.config.types.AttestationSubnets.deserialize(enr.get("attnets"))[subnet])
+        .map((enr: ENR) => enr.peerId().then((peerId) => {
+          const peerInfo = new PeerInfo(peerId);
+          peerInfo.multiaddrs.add(enr.multiaddrTCP);
+          return peerInfo;
+        })));
+    const peerInfos = peerInfosForSubnet.filter(peerInfo => !peerIds.includes(peerInfo.id.toB58String()));
+    // make sure they still connect to same subnet
+    let count = 0;
+    for (const peerInfo of peerInfos) {
+      // we'll dial thru sendRequest so don't need to do connect(peerInfo) like in the spec
+      const metadata = await this.reqResp.metadata(peerInfo);
+      if (metadata.attnets[subnet]) {
+        count++;
+      }
+      if (count < 10) {
```

Comment:
Do we need to limit the max peer to connect per subnet like this? If yes, need to specify a number.